numpy/core/tests/test_memmap.py

```diff
@@ -126,5 +128,23 @@ def test_view(self):
         new_array = asarray(fp)
         assert_(new_array.base is fp)
 
+    def test_ufunc_return_ndarray(self):
+        fp = memmap(self.tmpfp, dtype=self.dtype, shape=self.shape)
+        fp[:] = self.data
+
+        for unary_op in [sum, average, product]:
+            result = unary_op(fp)
+            assert_(isscalar(result))
+            assert_(result.__class__ is self.data[0, 0].__class__)
+
+            assert_(unary_op(fp, axis=0).__class__ is ndarray)
+            assert_(unary_op(fp, axis=1).__class__ is ndarray)
+
+        for binary_op in [add, multiply, divide, add]:
+            assert_(binary_op(fp, self.data).__class__ is ndarray)
```

Comment:
I suspect that the test failure is because `self.data` starts with zero, which causes problems with division.